aiida_mlip/parsers.py

```diff
@@ -1,58 +1,136 @@
+Singlepointcalc = CalculationFactory("janus.sp")
 
 
-class DiffParser(Parser):
+def convert_numpy(dictionary: dict) -> dict:
+    """
+    A function to convert numpy ndarrays in dictionary into lists.
+
+    Parameters
+    ----------
+    dictionary : dict
+        A dictionary with numpy array values to be converted into lists.
+
+    Returns
+    -------
+    dict
+        Converted dictionary.
+    """
+    for key, value in dictionary.items():
+        if isinstance(value, np.ndarray):
+            dictionary[key] = value.tolist()
+    return dictionary
+
+
+class SPParser(Parser):
     """
     Parser class for parsing output of calculation.
+
+    Parameters
+    ----------
+    node : aiida.orm.nodes.process.process.ProcessNode
+        ProcessNode of calculation.
+
+    Methods
+    -------
+    __init__(node: aiida.orm.nodes.process.process.ProcessNode)
+        Initialize the SPParser instance.
+
+    parse(**kwargs: Any) -> int:
+        Parse outputs, store results in the database.
+
+    Returns
+    -------
+    int
+        An exit code.
+
+    Raises
+    ------
+    exceptions.ParsingError
+        If the ProcessNode being passed was not produced by a Singlepointcalc.
     """
 
     def __init__(self, node):
         """
-        Initialize Parser instance
-
-        Checks that the ProcessNode being passed was produced by a DiffCalculation.
+        Check that the ProcessNode being passed was produced by a Singlepointcalc.
 
-        :param node: ProcessNode of calculation
-        :param type node: :class:`aiida.orm.nodes.process.process.ProcessNode`
+        Parameters
+        ----------
+        node : aiida.orm.nodes.process.process.ProcessNode
+            ProcessNode of calculation.
         """
         super().__init__(node)
-        if not issubclass(node.process_class, DiffCalculation):
-            raise exceptions.ParsingError("Can only parse DiffCalculation")
+        if not issubclass(node.process_class, Singlepointcalc):
+            raise exceptions.ParsingError("Can only parse Singlepointcalc")
 
+    # pylint: disable=too-many-locals
     def parse(self, **kwargs):
         """
-        Parse outputs, store results in database.
+        Parse outputs, store results in the database.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Any keyword argument.
 
-        :returns: an exit code, if parsing fails (or nothing if parsing succeeds)
+        Returns
+        -------
+        int
+            An exit code.
         """
         output_filename = self.node.get_option("output_filename")
+        xyzoutput = (self.node.inputs.xyz_output_name).value
+        logoutput = (self.node.inputs.log_filename).value
+
+        remote_folder = self.node.get_remote_workdir()
 
         # Check that folder content is as expected
         files_retrieved = self.retrieved.list_object_names()
-        files_expected = [output_filename]
+
+        files_expected = [xyzoutput, logoutput]
```

Comment:
Instead of casting to `set` you can just construct a `set`
```suggestion
        files_expected = {xyzoutput, logoutput}
```

Reply:
This is also single use and could be a constant in the `if` below